segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/SegmentKeyCache.java

```diff
@@ -42,20 +43,21 @@
  * The cache is separated in this manner because the Tail Section is optimized to bear the brunt of all Index Modifications
  * to a Table Segment; all updates and removals will end up modifying this section directly, so it is important that it
  * provides an easily modifiable data structure. The Index Section is designed for less frequent updates but is can handle
- * a larger amount of data being cached (since it is backed by the process-wide {@link Cache}). The Tail Section, while
+ * a larger amount of data being cached (since it is backed by the process-wide {@link CacheStorage}). The Tail Section, while
  * dynamic, is not expected to grow too large due to the Table Segment being continuously indexed in the background, which
  * causes the Last Indexed Offset to be updated frequently.
  */
 @ThreadSafe
 @RequiredArgsConstructor
 class SegmentKeyCache {
     //region Members
+    private static final int HASH_GROUP_COUNT = 1024;
```

Comment:
What's the real impact of making this constant larger or smaller?

Reply:
This is the number of Cache Entries that the Segment Key Cache will use. Making it larger will increase the number of Cache Entries used, but they will be shorter on average, while making them smaller will decrease the number of cache entries used but storing more in each. I decided to decrease this because it is now much cheaper to read/write cache entries (even if we have more than one entry), but we'll make better use of the cache space with bigger entries due to the 4KB block size.